cjdns-entities/src/lib.rs

```diff
@@ -177,17 +183,43 @@ impl fmt::Display for Label64 {
     }
 }
 
+impl fmt::Display for Label128 {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{:04x}.{:04x}.{:04x}.{:04x}.{:04x}.{:04x}.{:04x}.{:04x}",
+            (self.0 >> 112) & 0xFFFFu128,
+            (self.0 >> 96) & 0xFFFFu128,
+            (self.0 >> 80) & 0xFFFFu128,
+            (self.0 >> 64) & 0xFFFFu128,
+            (self.0 >> 48) & 0xFFFFu128,
+            (self.0 >> 32) & 0xFFFFu128,
+            (self.0 >> 16) & 0xFFFFu128,
+            self.0 & 0xFFFFu128
+        )
+    }
+}
+
 fn capture2u64(c: &regex::Captures, group_num: usize) -> u64 {
     u64::from_str_radix(c.get(group_num).unwrap().as_str(), 16).unwrap()
 }
 
+fn capture2u128(c: &regex::Captures, group_num: usize) -> u128 {
```

Comment:
Since the groups are small (2 bytes), extra fn is not needed. Use `capture2u64(...) as u128` below.